provider/pkg/gen/python-templates/yaml/yaml.tmpl

```diff
@@ -3,17 +3,214 @@
+               transformations to apply to Kubernetes resource definitions before registering with engine.
+        :param Optional[str] resource_prefix: An optional prefix for the auto-generated resource names.
+               Example: A resource created with resource_prefix="foo" would produce a resource named "foo-resourceName".
+        """
+        if not name:
+            raise TypeError('Missing resource name argument (for URN creation)')
+        if not isinstance(name, str):
+            raise TypeError('Expected resource name to be a string')
+        if opts and not isinstance(opts, pulumi.ResourceOptions):
+            raise TypeError('Expected resource options to be a ResourceOptions instance')
+        if not files:
+            files = []
+        if not yaml:
+            yaml = []
+
+        __props__ = dict()
+
+        if resource_prefix:
+            name = f"{resource_prefix}-{name}"
+        super(ConfigGroup, self).__init__(
+            "kubernetes:yaml:ConfigGroup",
+            name,
+            __props__,
+            opts)
+
+        self.resources = pulumi.Output.from_input({})
+
+        _files: List[str] = []
+        for file in files:
+            if _is_url(file):
+                _files.append(file)
+            else:
+                _files += [f for f in glob(file)]
+
+        opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(parent=self))
+
+        for file in _files:
+            cf = ConfigFile(
+                file, file_id=file, transformations=transformations, resource_prefix=resource_prefix, opts=opts)
+            # Add any new ConfigFile resources to the ConfigGroup's resources
+            self.resources = pulumi.Output.all(cf.resources, self.resources).apply(lambda x: {**x[0], **x[1]})
+
+        for text in yaml:
+            # Rather than using the default provider for the following invoke call, use the version specified
+            # in package.json.
+            invoke_opts = pulumi.InvokeOptions(version=get_version())
+
+            __ret__ = pulumi.runtime.invoke('kubernetes:yaml:decode', {'text': text}, invoke_opts).value['result']
+            resources = _parse_yaml_document(__ret__, opts, transformations, resource_prefix)
+            # Add any new YAML resources to the ConfigGroup's resources
+            self.resources = pulumi.Output.all(resources, self.resources).apply(lambda x: {**x[0], **x[1]})
+
+        # Note: Unlike NodeJS, Python requires that we "pull" on our futures in order to get them scheduled for
+        # execution. In order to do this, we leverage the engine's RegisterResourceOutputs to wait for the
+        # resolution of all resources that this YAML document created.
+        self.register_outputs({"resources": self.resources})
+
+    def translate_output_property(self, prop: str) -> str:
+        return tables._CAMEL_TO_SNAKE_CASE_TABLE.get(prop) or prop
```

Comment:
If you pull in the latest codegen, this will become `_tables.CAMEL_CASE_TO_SNAKE_CASE_TABLE.get(prop)`. It's not specifically relevant to this PR but just wanted to mention that upcoming change.